src/legacy/core_plugins/embeddable_api/public/np_ready/public/lib/panel/panel_header/panel_actions/add_panel/change_view_flyout.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { CoreSetup } from 'src/core/public';
+import { DashboardPanelState } from 'src/legacy/core_plugins/dashboard_embeddable_container/public/np_ready/public';
+
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  // @ts-ignore
+  EuiSuperSelect,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { IContainer } from '../../../../containers';
+import { GetEmbeddableFactory, GetEmbeddableFactories } from '../../../../types';
+import { IEmbeddable, EmbeddableInput, EmbeddableOutput } from '../../../../embeddables';
+
+interface Props {
+  onClose: () => void;
+  container: IContainer;
+  getFactory: GetEmbeddableFactory;
+  getAllFactories: GetEmbeddableFactories;
+  notifications: CoreSetup['notifications'];
+  SavedObjectFinder: React.ComponentType<any>;
+  viewToRemove: IEmbeddable<EmbeddableInput, EmbeddableOutput>;
+}
+
+export class ChangeViewFlyout extends React.Component<Props> {
+  private lastToast: any;
+
+  constructor(props: Props) {
+    super(props);
+  }
+
+  public showToast = (name: string) => {
+    // To avoid the clutter of having toast messages cover flyout
+    // close previous toast message before creating a new one
+    if (this.lastToast) {
+      this.props.notifications.toasts.remove(this.lastToast);
+    }
+
+    this.lastToast = this.props.notifications.toasts.addSuccess({
+      title: i18n.translate(
+        'embeddableApi.addPanel.savedObjectAddedToContainerSuccessMessageTitle',
+        {
+          defaultMessage: '{savedObjectName} was added',
+          values: {
+            savedObjectName: name,
+          },
+        }
+      ),
+      'data-test-subj': 'addObjectToContainerSuccess',
+    });
+  };
+
+  public onChangeView = async (id: string, type: string, name: string) => {
+    const originalPanels = this.props.container.getInput().panels;
+    const filteredPanels = { ...originalPanels };
+
+    const nnw = (filteredPanels[this.props.viewToRemove.id] as DashboardPanelState).gridData.w;
```

Comment:
Ah, I just realized this action is coupled to the dashboard container.  In that case it should actually not be a part of this plugin but the `dashboard_container_embeddable` one.  We don't want it to show up for embeddables that aren't in a dashboard, and this plugin doesn't depend on that one (other way around actually).

When you do that you'll have to attach this action to the `CONTEXT_MENU_TRIGGER`, similar to the other action dashboard registers:

```
    uiActions.attachAction(CONTEXT_MENU_TRIGGER, expandPanelAction.id);
```

Reply:
Sorry, it's not clear to me.
All the other actions (add panel, inspect panel, etc.) are under the embeddable_api subtree.
I just placed my action with those other ones (maybe wrongly).

Reply:
The actions in the embeddable_api are relevant for all embeddables, in any container.  If an action is only applicable to a certain type, it should be in that plugin.  If you check out `ExpandPanelAction` inside the `dashboard_embeddable_container` plugin, you can use that as an example.  The way it is now there is a circular dependency - you are importing code from the dashboard embeddable plugin, but that plugin needs code from this plugin.  This won't work when we migrate all these plugins to the "new platform".

lmk if you need some more help on how to migrate this action over!

Reply:
Hope to have done this right...